webview_panels/src/modules/documentationEditor/components/settings/DocGeneratorSettings.tsx

```diff
@@ -39,47 +36,45 @@ const DocGeneratorSettings = (): JSX.Element => {
   };
 
   return (
-    <>
-      <Button outline onClick={() => setShowSettingsPanel(true)}>
-        <SettingsIcon style={{ height: 16 }} /> Settings
-      </Button>
-      {showSettingsPanel ? (
-        <RightSidePanel
-          title="Settings"
-          onClose={() => setShowSettingsPanel(false)}
-        >
-          <Stack direction="column">
-            <h5>Configure settings for document generation</h5>
-            <Stack direction="column">
-              <Label>
-                Language
-                <Select
-                  name="language"
-                  defaultValue={{
-                    label: userInstructions.language ?? DefaultLanguage,
-                    value: userInstructions.language ?? DefaultLanguage,
-                  }}
-                  onChange={handleChange}
-                  options={Languages.map((l) => ({ label: l, value: l }))}
-                />
-              </Label>
-              <Label>
-                Persona
-                <Select
-                  name="persona"
-                  defaultValue={{
-                    label: userInstructions.persona ?? DefaultPersona,
-                    value: userInstructions.persona ?? DefaultPersona,
-                  }}
-                  onChange={handleChange}
-                  options={Persona.map((l) => ({ label: l, value: l }))}
-                />
-              </Label>
-            </Stack>
-          </Stack>
-        </RightSidePanel>
-      ) : null}
-    </>
+    <Drawer
```

Comment:
Used common `Drawer` component to handle functions like keyboard escape, clicking on backdrop etc